src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4485,15 +4485,22 @@ trait Types
 
   /** Do type arguments `targs` conform to formal parameters `tparams`?
    */
-  def isWithinBounds(pre: Type, owner: Symbol, tparams: List[Symbol], targs: List[Type]): Boolean = {
-    var bounds = instantiatedBounds(pre, owner, tparams, targs)
-    if (targs exists typeHasAnnotations)
+  def isWithinBounds(pre: Type, owner: Symbol, tparams: List[Symbol], targs: List[Type]): Boolean =
+    if (targs exists typeHasAnnotations){
+      var bounds = instantiatedBounds(pre, owner, tparams, targs)
       bounds = adaptBoundsToAnnotations(bounds, tparams, targs)
-    (bounds corresponds targs)(boundsContainType)
-  }
+      (bounds corresponds targs)(boundsContainType)
+    } else
+      (tparams corresponds targs){ (tparam, targ) =>
+        val bound = instantiatedBound(pre, owner, tparams, targs, tparam)
+        boundsContainType(bound, targ)
+      }
 
   def instantiatedBounds(pre: Type, owner: Symbol, tparams: List[Symbol], targs: List[Type]): List[TypeBounds] =
```

Comment:
@retronym Question: this method `instantiatedBounds` is only used from `isWithinBounds`, so we could put it into a local method there. However, I am not sure if, since this is a public method of the `Types` trait, we can remove it safely; and since the trait `Types` is the `scala.reflect.internal` package... Could you clarify?

Reply:
The advantage of pulling against 2.13 is to avoid that issue, but in this case it must be moot. I'd even be curious to see what breaks in the community build, to see who thought, Oh there's an interesting method.